lib/src/arithmetic/floating_point/floating_point_values/floating_point_32_value.dart

```diff
@@ -0,0 +1,89 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// floating_point_32_value.dart
+// Implementation of 32-bit Floating-Point value representations.
+//
+// 2024 October 15
+// Authors:
+//  Max Korbel <max.korbel@intel.com>
+//  Desmond A Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'dart:typed_data';
+import 'package:meta/meta.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// A representation of a single-precision floating-point value.
+class FloatingPoint32Value extends FloatingPointValue {
+  /// The exponent width
+  static const int exponentWidth = 8;
+
+  /// The mantissa width
+  static const int mantissaWidth = 23;
+
+  @override
+  @protected
+  int get constrainedExponentWidth => exponentWidth;
+
+  @override
+  @protected
+  int get constrainedMantissaWidth => mantissaWidth;
+
+  /// Constructor for a single precision floating point value
+  FloatingPoint32Value(
+      {required super.sign, required super.exponent, required super.mantissa});
+
+  /// Return the [FloatingPoint32Value] representing the constant specified
+  factory FloatingPoint32Value.getFloatingPointConstant(
+          FloatingPointConstants constantFloatingPoint) =>
+      FloatingPoint32Value.fromLogicValue(
+          FloatingPointValue.getFloatingPointConstant(
+                  constantFloatingPoint, exponentWidth, mantissaWidth)
+              .value);
+
+  /// [FloatingPoint32Value] constructor from string representation of
+  /// individual bitfields
+  FloatingPoint32Value.ofBinaryStrings(
+      super.sign, super.exponent, super.mantissa)
+      : super.ofBinaryStrings();
+
+  /// [FloatingPoint32Value] constructor from spaced string representation of
+  /// individual bitfields
+  FloatingPoint32Value.ofSpacedBinaryString(super.fp)
+      : super.ofSpacedBinaryString();
+
+  /// [FloatingPoint32Value] constructor from a single string representing
+  /// space-separated bitfields
+  FloatingPoint32Value.ofString(String fp, {super.radix})
+      : super.ofString(fp, exponentWidth, mantissaWidth);
+
+  /// [FloatingPoint32Value] constructor from a set of [BigInt]s of the binary
+  /// representation
+  FloatingPoint32Value.ofBigInts(super.exponent, super.mantissa, {super.sign})
+      : super.ofBigInts();
+
+  /// [FloatingPoint32Value] constructor from a set of [int]s of the binary
+  /// representation
+  FloatingPoint32Value.ofInts(super.exponent, super.mantissa, {super.sign})
+      : super.ofInts();
+
+  /// Numeric conversion of a [FloatingPoint32Value] from a host double
+  factory FloatingPoint32Value.fromDouble(double inDouble) {
+    final byteData = ByteData(4)..setFloat32(0, inDouble);
```

Comment:
This code could be moved to parent method as in the other floating point types.

Reply:
only fp32 and fp64 have this code, and they're not quite identical.  is that what you mean?

Reply:
Yes, 32 and 64 exploiting native ways. It is fine to keep here.